blogs/2023/07/20/mangling-vscode.md

```diff
@@ -0,0 +1,177 @@
+
+While debugging our minified source code on vscode.dev last year though, I noticed something surprising: our minified JavaScript still included tons of long identifier names, such as `extensionIgnoredRecommendationsService`. This surprised me. I assumed esbuild would have already shortened these identifiers. And it turns out esbuild actually does shorten identifiers in some cases through a process called "mangling" (a term JavaScript tools likely borrowed from an [only roughly similar process for compiled languages](https://en.wikipedia.org/wiki/Name_mangling)).
+
+During minification, mangling shortens long identifier names, transforming code such as:
+
+```ts
+const someLongVariableName = 123;
+console.log(someLongVariableName);
+```
+
+To the much shorter:
+
+```ts
+const x = 123;
+console.log(x);
+```
+
+Since JavaScript is shipped as source text, reducing the length of identifier names actually decreases the program's size. I know that optimization probably seems more than a little silly if you're coming from a compiled language, but here in the wonderful world of JavaScript we gladly take wins like this wherever we can find them!
+
+Now before you rush out to rename all of your variables to single letters, I want to stress that optimizations like this need to be approached cautiously. If a potential optimization makes your source code less readable or maintainable, or requires significant manual work, it's almost never worth it unless it delivers truly spectacular improvements. Shaving off a few bytes here and there is nice but hardly qualifies as spectacular.
+
+That calculus changes if we can get nice optimizations like this essentially for free, say by having our build tool do them for us automatically. And indeed, smart tools like esbuild already implement identifier mangling. That means we can keep writing our `veryLongAndDescriptiveNamesThatWouldMakeEvenObjectiveCProgrammersBlush` and let our build tools shorten them for us!
+
+Even though esbuild implements mangling, by default it only mangles names when it is reasonably confident that mangling won't change the behavior of the code. After all, having a bundler break your code really stinks. In practice, this means that esbuild mangles local variable names and argument names. This is safe unless your code is doing some truly crazy stuff (in which case, you likely have far bigger issues than code size to worry about).
```

Comment:
Not "reasonably confident" but absolutely confident. Therefore it only tackles a subset of the identifies for which is "compute" such confidence

Reply:
I'll make this stronger. I wanted to leave a little wiggle room as there are still some terrible cases involving `eval` or `Function.toString` that esbuild can break